project_redss/apply_manual_codes.py

```diff
@@ -1,127 +1,132 @@
 from core_data_modules.util import IOUtils
 from dateutil.parser import isoparse
 
+from project_redss.lib import MessageFilters
 from project_redss.lib.dataset_specification import DatasetSpecification
+from project_redss.lib.redss_schemes import CodeSchemes
 
 
 class ApplyManualCodes(object):
-    VARIABLE_NAME = "S07E01_Humanitarian_Priorities"
-    FLOW_NAME = "esc4jmcna_activation"
-
     @classmethod
-    def apply_manual_codes(cls, user, data, coded_input_path, interface_output_dir):
-        # Merge manually coded survey/evaluation Coda files into the cleaned dataset
-        for plan in DatasetSpecification.coding_plans:
-            coda_file_path = path.join(coded_input_path, "{}_coded.csv".format(plan.coda_name))
-
-            if not path.exists(coda_file_path):
-                print("Warning: No Coda file found for key '{}'".format(plan.coda_name))
-                for td in data:
-                    td.append_data(
-                        {plan.manually_coded_field: Codes.NOT_REVIEWED},
-                        Metadata(user, Metadata.get_call_location(), time.time())
-                    )
-                continue
-
-            with open(coda_file_path, "r") as f:
-                TracedDataCodaIO.import_coda_to_traced_data_iterable(
-                    user, data, plan.source_field, {plan.coda_name: plan.manually_coded_field}, f, True)
-
-        # Set districts coded as 'other' to 'NOT_CODED'
-        for td in data:
-            if td["district_coded"] == "other":
-                td.append_data({"district_coded": Codes.NOT_CODED},
-                               Metadata(user, Metadata.get_call_location(), time.time()))
+    def apply_manual_codes(cls, user, data, coda_input_dir, interface_output_dir):
+        # Merge manually coded radio show files into the cleaned dataset
+        for plan in DatasetSpecification.RQA_CODING_PLANS:
+            rqa_messages = [td for td in data if plan.raw_field in td]
+
+            nr_label = CleaningUtils.make_label(
+                plan.code_scheme, plan.code_scheme.get_code_with_control_code(Codes.NOT_REVIEWED),
+                Metadata.get_call_location()
+            )
+
+            coda_input_path = path.join(coda_input_dir, "{}.json".format(plan.coda_filename))
+            if path.exists(coda_input_path):
+                with open(coda_input_path, "r") as f:
+                    TracedDataCoda2IO.import_coda_2_to_traced_data_iterable_multi_coded(
+                        user, rqa_messages, plan.id_field, {plan.coded_field: plan.code_scheme.scheme_id}, nr_label, f)
+            else:
+                # Read from simulated empty file
+                TracedDataCoda2IO.import_coda_2_to_traced_data_iterable_multi_coded(
+                    user, rqa_messages, plan.id_field, {plan.coded_field: plan.code_scheme.scheme_id}, nr_label,
+                    io.StringIO("[]"))
```

Comment:
Could you move this simulated empty file into the core library as a constant please